apps/voting/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+function useIdentity(address) {
+  const [name, setName] = React.useState(null)
+  const { resolve, updates$, showLocalIdentityModal } = React.useContext(
+    IdentityContext
+  )
+
+  const handleNameChange = metadata => {
+    setName(metadata ? metadata.name : null)
+  }
+
+  const handleShowLocalIdentityModal = address => {
+    // Emit an event whenever the modal is closed (when the promise resolves)
+    return showLocalIdentityModal(address)
+      .then(() => updates$.next(address))
+      .catch(e => null)
+  }
+
+  React.useEffect(() => {
+    resolve(address).then(handleNameChange)
+
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        // Resolve and update state when the identity have been updated
+        resolve(address).then(handleNameChange)
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [address])
+
+  return [name, handleShowLocalIdentityModal]
+}
+
+const LocalIdentityBadge = ({ address, ...props }) => {
+  const [label, showLocalIdentityModal] = useIdentity(address)
+  const handleClick = () => showLocalIdentityModal(address)
+  return (
+    <IdentityBadge
+      {...props}
+      customLabel={label || ''}
+      entity={address}
+      popoverAction={{
+        label: `${label ? 'Edit' : 'Add'} custom label`,
+        onClick: handleClick,
+      }}
+      popoverTitle={
+        label ? (
+          <Wrap>
+            <Address>{label}</Address>
+            <StyledBadge>Custom label</StyledBadge>
+          </Wrap>
+        ) : (
+          'Address'
+        )
+      }
+    />
+  )
+}
+
+LocalIdentityBadge.propTypes = {
+  address: PropTypes.string.isRequired,
+}
+
+const Wrap = styled.div`
+  display: grid;
+  align-items: center;
+  grid-template-columns: auto 1fr;
+  padding-right: 24px;
+`
+
+const Address = styled.span`
```

Comment:
Perhaps this should be named `Label` instead (it would never actually be used if just an address was given)?